opm/autodiff/WellHelpers.hpp

```diff
@@ -191,6 +191,63 @@ namespace Opm {
             return retval;
         }
 
+        template <class MatrixType>
+        bool invertMatrixLapack(MatrixType& matrix) {
+
+            const int m = matrix.M();
+            const int n = matrix.N();
```

Comment:
Do recall that `matrix.N()` returns the number of **rows** and that `matrix.M()` returns the number of **columns** in the case of a `Dune::FieldMatrix<>`.  We do assert that the two are equal (maybe that should be an exception instead), but at least in light of `m` being used for row indices and `n` being used for column indices that could be confusing.

Reply:
> Do recall that matrix.N() returns the number of rows and that matrix.M() returns the number of columns in the case of a Dune::FieldMatrix<>. We do assert that the two are equal (maybe that should be an exception instead), but at least in light of m being used for row indices and n being used for column indices that could be confusing.

Yes. `m` and `n` should have better names. `matrix.M()` does return the number of `rows` and `matrix.N()` returns the number of `columns`.  An exception is also good.

Reply:
> `matrix.M()` does return the number of rows and `matrix.N()` returns the number of columns

If so, that's a recent breaking change in Dune.  At least up to and including 2.4.1 `matrix.N()` returned the number of **rows** and `matrix.M()` returned the number of **columns**.

Reply:
> f so, that's a recent breaking change in Dune. At least up to and including 2.4.1 matrix.N() returned the number of rows and matrix.M() returned the number of columns.

You are right, I misread it.